src/Query.php

```diff
@@ -19,18 +20,62 @@
  * @author Nicolas Ruflin <spam@ruflin.com>
  *
  * @see https://www.elastic.co/guide/en/elasticsearch/reference/current/search-request-body.html
+ * @phpstan-type TCreateQueryArgs = TCreateQueryArgsForModifyingQueries|AbstractSuggest|Collapse|Suggest
+ * @phpstan-type TCreateQueryArgsForModifyingQueries = AbstractQuery|TRawQuery|self|string|null
+ * @phpstan-type TRawQuery = array{
```

Comment:
Haven't really used phpstan on my end yet. How will we make sure this stays up-to-date? Is this automated / validated by phpstan? If it is automated, do we actually need it in the source code or could the IDE do this directly? 

My more fundamental question is: What problem are we solving with this addition / how will it make the developer experience better.

Reply:
> How will we make sure this stays up-to-date? Is this automated / validated by phpstan?

It's not automated and must be edited by hand.

Yes, it's validated:
1. you can't write a piece of code that violate annotations
2. you can't write an annotation that violate existing code

And the validation is already in the CI: https://github.com/ruflin/Elastica/runs/7645330172?check_suite_focus=true#step:7:1

> What problem are we solving with this addition / how will it make the developer experience better.

The problem is that `array` doesn't have/give any information about its contents. Adding enhanced type annotations help to understand what a piece of code does, what it accepts and what it produces.